tornado/http1connection.py

```diff
@@ -598,6 +598,7 @@ def _read_chunked_body(self, delegate):
             chunk_len = yield self.stream.read_until(b"\r\n", max_bytes=64)
             chunk_len = int(chunk_len.strip(), 16)
             if chunk_len == 0:
+                yield self.stream.read_bytes(2) # comsume tailing \r\n, fix bug for multiple chunked requests on same connection.
```

Comment:
We should assert that the two bytes are in fact `\r\n`: 

    crlf = yield self.stream.read_bytes(2)
    if crlf != b'\r\n':
        raise HTTPInputError("improperly terminated chunked request")

And it looks like we have some tests that need to be fixed up to send the proper format.